adapters/http-base/src/main/java/org/eclipse/hono/adapter/http/JwtAuthHandler.java

```diff
@@ -0,0 +1,182 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.adapter.http;
+
+import java.net.HttpURLConnection;
+import java.util.Objects;
+
+import org.eclipse.hono.adapter.HttpContext;
+import org.eclipse.hono.adapter.auth.device.DeviceCredentialsAuthProvider;
+import org.eclipse.hono.adapter.auth.device.ExecutionContextAuthHandler;
+import org.eclipse.hono.adapter.auth.device.PreCredentialsValidationHandler;
+import org.eclipse.hono.adapter.auth.device.jwt.CredentialsParser;
+import org.eclipse.hono.adapter.auth.device.jwt.DefaultJwsValidator;
+import org.eclipse.hono.adapter.auth.device.jwt.JwtCredentials;
+import org.eclipse.hono.client.ClientErrorException;
+import org.eclipse.hono.client.ServiceInvocationException;
+import org.eclipse.hono.util.CredentialsConstants;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.MalformedJwtException;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.User;
+import io.vertx.ext.auth.authentication.AuthenticationProvider;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.HttpException;
+import io.vertx.ext.web.handler.impl.HTTPAuthorizationHandler;
+
+/**
+ * A Hono specific version of vert.x web's standard {@code JWTAuthHandlerImpl}.
```

Comment:
does this mean that the code in this class has been copied from vert.x? If so, we need to maintain the original license header of that  class and amend it with a description of what you have changed, instead of putting in our own header.

Reply:
Since we removed the only logic that was left from the original vert.x version (see comment below) I think we can get rid of this reference. What do you think?

Reply:
Sounds reasonable